src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -96,19 +96,20 @@ def _msal_app(self):
         msal_authority = "https://{}/{}".format(self.authority, self.tenant_id)
         return PublicClientApplication(authority=msal_authority, client_id=self.client_id, token_cache=cache)
 
-    def login_with_interactive_browser(self):
+    def login_with_interactive_browser(self, scopes=None):
```

Comment:
If multiple methods requires `scopes`, how about put it in `Identity.__init__`?

Reply:
I was thinking about the same. I did not want to change the "state" of the `Identity` object with the `scopes` member variable because I did not know whether it is expected or not. As you mentioned this here, I am super happy to go into this direction, going to do that.

Reply:
I have push a new commit with this change. Looking forward to your opinion.